src/server/cluster/cluster_config.cc

```diff
@@ -278,6 +311,10 @@ optional<ClusterConfig::ClusterShards> BuildClusterConfigFromJson(const JsonType
       shard.replicas.push_back(std::move(node).value());
     }
 
+    if (auto migrations = ParseMigrations(element.at_or_null("migrations")); migrations) {
```

Comment:
If you have an error in ParseMigrations you should return nullopt here which will cause cluster config command to return error